single_step_avd/src/merkle_tree_avd/store/mem_store/mod.rs

```diff
@@ -0,0 +1,72 @@
+use std::collections::HashMap;
+use crate::{
+    Error,
+};
+use crypto_primitives::{
+    sparse_merkle_tree::{
+        store::SMTStorer,
+        MerkleIndex,
+        MerkleTreeParameters,
+        SparseMerkleTree,
+    },
+    hash::FixedLengthCRH,
+};
+use crate::merkle_tree_avd::{
+    concat_leaf_data,
+    MerkleTreeAVDParameters,
+    store::MTAVDStorer,
+};
+
+pub struct MTAVDMemStore<P: MerkleTreeAVDParameters> {
```

Comment:
@nirvantyagi hey I don't understand this compiler warning:

the function or associated item `new` exists for struct `merkle_tree_avd::store::mem_store::MTAVDMemStore<_>`, but its trait bounds were not satisfied function or associated item `new` not found for this note: the following trait bounds were not satisfied: `merkle_tree_avd::store::mem_store::MTAVDMemStore<_>: sha3::digest::Update` which is required by `merkle_tree_avd::store::mem_store::MTAVDMemStore<_>: sha3::Digest` `merkle_tree_avd::store::mem_store::MTAVDMemStore<_>: sha3::digest::FixedOutput` which is required by `merkle_tree_avd::store::mem_store::MTAVDMemStore<_>: sha3::Digest` `merkle_tree_avd::store::mem_store::MTAVDMemStore<_>: sha3::digest::Reset` which is required by `merkle_tree_avd::store::mem_store::MTAVDMemStore<_>: sha3::Digest` `merkle_tree_avd::store::mem_store::MTAVDMemStore<_>: std::clone::Clone` which is required by `merkle_tree_avd::store::mem_store::MTAVDMemStore<_>: sha3::Digest` `merkle_tree_avd::store::mem_store::MTAVDMemStore<_>: std::default::Default` which is required by `merkle_tree_avd::store::mem_store::MTAVDMemStore<_>: sha3::Digest` `&merkle_tree_avd::store::mem_store::MTAVDMemStore<_>: sha3::digest::Update` which is required by `&merkle_tree_avd::store::mem_store::MTAVDMemStore<_>: sha3::Digest` `&merkle_tree_avd::store::mem_store::MTAVDMemStore<_>: sha3::digest::FixedOutput` which is required by `&merkle_tree_avd::store::mem_store::MTAVDMemStore<_>: sha3::Digest` `&merkle_tree_avd::store::mem_store::MTAVDMemStore<_>: sha3::digest::Reset` which is required by `&merkle_tree_avd::store::mem_store::MTAVDMemStore<_>: sha3::Digest` `&merkle_tree_avd::store::mem_store::MTAVDMemStore<_>: std::default::Default` which is required by `&merkle_tree_avd::store::mem_store::MTAVDMemStore<_>: sha3::Digest` `&mut merkle_tree_avd::store::mem_store::MTAVDMemStore<_>: sha3::digest::Update` which is required by `&mut merkle_tree_avd::store::mem_store::MTAVDMemStore<_>: sha3::Digest` `&mut merkle_tree_avd::store::mem_store::MTAVDMemStore<_>: sha3::digest::FixedOutput` which is required by `&mut merkle_tree_avd::store::mem_store::MTAVDMemStore<_>: sha3::Digest` `&mut merkle_tree_avd::store::mem_store::MTAVDMemStore<_>: sha3::digest::Reset` which is required by `&mut merkle_tree_avd::store::mem_store::MTAVDMemStore<_>: sha3::Digest` `&mut merkle_tree_avd::store::mem_store::MTAVDMemStore<_>: std::clone::Clone` which is required by `&mut merkle_tree_avd::store::mem_store::MTAVDMemStore<_>: sha3::Digest` `&mut merkle_tree_avd::store::mem_store::MTAVDMemStore<_>: std::default::Default` which is required by `&mut merkle_tree_avd::store::mem_store::MTAVDMemStore<_>: sha3::Digest` help: items from traits can only be used if the trait is in scope help: consider annotating `merkle_tree_avd::store::mem_store::MTAVDMemStore<_>` with `#[derive(std::clone::Clone, std::default::Default)]`: `#[derive(std::clone::Clone, std::default::Default)] `